lib/Makefile

```diff
@@ -41,6 +41,10 @@ CPPFLAGS  += -I./legacy -DZSTD_LEGACY_SUPPORT=1
 ZSTD_FILES+= $(wildcard legacy/*.c)
 endif
 
+ifeq ($(shell uname), FreeBSD)
+CPPFLAGS  += -D_POSIX_C_SOURCE=200112L
```

Comment:
It seems this detection macro is not useful in `/lib`.
Feel free to explain if you believe differently.